### PR TITLE
#51 - Admin Panel for Quizzes: Quiz Update API

### LINKED ISSUE #51: Admin Panel for Quizzes: Quiz Update API
Provide a `PATCH`  endpoint that allows frontend to send `Quiz`, `Question`  and `Answer` models' data to the backend as one object.
The example data sent via request:
```JSON
{
  "id": 0,
  "name": "Quiz Name",
  "scheduleAt": "2024-08-28T15:00:00.000000Z",
  "scheduleUntil": "2025-08-28T15:00:00.000000Z",
  "questions": [
    {
      "id": 0,
      "text": "Question's content",
      "answers": [
        {
          "id": 0,
          "text": "Answer's content",
          "correct": true
        }
      ]
    }
  ]
}

```

### PR DESCRIPTION
**Pull request by AmonDeShir**
Should close #51.

### CODE REVIEW

File: app/Models/Question.php [hunk 1]
```diff
@@ -75,4 +75,31 @@ public function cloneTo(Quiz $quiz): self
...
+            if (array_key_exists("id", $answerData)) {
+                $answer = $this->answers()->findOrFail($answerData["id"]);
+            } else {
+                $answer = $this->answers()->create($answerData);
+            }
+
+            if (array_key_exists("correct", $answerData)) {
+                $this->correctAnswer()->associate($answer);
+            }
+
```
Comment (kamilpiech97) [inline app/Models/Question.php:99]: Lest move that into some service.

File: app/Models/Quiz.php [hunk 2]
```diff
@@ -97,6 +97,29 @@ public function createSubmission(User $user): QuizSubmission
...
+            $question = null;
+
+            if (array_key_exists("id", $questionData)) {
+                $question = $this->questions()->findOrFail($questionData["id"]);
+            } else {
+                $question = $this->questions()->create($questionData);
+            }
+
+            $question->deepUpdate($questionData);
+        }
```
Comment (kamilpiech97) [inline app/Models/Quiz.php:118]: Lest move that into some service.

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: app/Models/Question.php (hunk 1 above)
Comment (AmonDeShir) [reply app/Models/Question.php]: done

File: app/Models/Quiz.php (hunk 2 above)
Comment (AmonDeShir) [reply app/Models/Quiz.php]: done

**Review by kamilpiech97 — APPROVED** [review]

**Review by AleksandraKozubal — APPROVED** [review]

**Pull request merged**